hail/python/hailtop/hailctl/batch/submit.py

```diff
@@ -1,108 +1,188 @@
-
-    def cloud_prefix(path):
-        path = path.lstrip('/')
-        return f'{remote_tmpdir}/{tmpdir_path_prefix}/{path}'
-
-    def file_input_to_src_dest(file: str) -> Tuple[str, str, str]:
-        match = FILE_REGEX.match(file)
-        if match is None:
-            raise ValueError(f'invalid file specification {file}. Must have the form "src" or "src:dest"')
-
-        result = match.groupdict()
-
-        src = result.get('src')
-        if src is None:
-            raise ValueError(f'invalid file specification {file}. Must have a "src" defined.')
-        src = os.path.abspath(os.path.expanduser(src))
-        src = src.rstrip('/')
-
-        dest = result.get('dest')
-        if dest is not None:
-            dest = os.path.abspath(os.path.expanduser(dest))
+        if output == 'text':
+            deploy_config = get_deploy_config()
+            url = deploy_config.external_url('batch', f'/batches/{bc_b.id}/jobs/1')
+            print(f'Submitted batch {bc_b.id}, see {url}')
+        # FIXME: support yaml
         else:
-            dest = os.getcwd()
-
-        cloud_file = cloud_prefix(src)
-
-        return (src, dest, cloud_file)
-
-    backend = hb.ServiceBackend()
-    b = hb.Batch(name=name, backend=backend)
-    j = b.new_bash_job()
-    j.image(image_name or os.environ.get('HAIL_GENETICS_HAIL_IMAGE', f'hailgenetics/hail:{__pip_version__}'))
-
-    local_files_to_cloud_files = []
+            assert output == 'json'
+            print(orjson.dumps({'id': bc_b.id}).decode('utf-8'))
 
-    for file in files:
-        src, dest, cloud_file = file_input_to_src_dest(file)
-        local_files_to_cloud_files.append({'from': src, 'to': cloud_file})
-        in_file = b.read_input(cloud_file)
-        j.command(f'mkdir -p {os.path.dirname(dest)}; ln -s {in_file} {dest}')
+        if wait:
+            bc_b.wait(disable_progress_bar=quiet)
 
-    script_src, _, script_cloud_file = file_input_to_src_dest(script)
-    user_config_src, _, user_config_cloud_file = file_input_to_src_dest(user_config)
 
-    await copy_from_dict(files=local_files_to_cloud_files)
-    await copy_from_dict(
-        files=[
-            {'from': script_src, 'to': script_cloud_file},
-            {'from': user_config_src, 'to': user_config_cloud_file},
-        ]
-    )
+async def get_user_config_files(fs: RouterAsyncFS) -> List[str]:
+    user_config_path = get_hail_config_path()
+    if not user_config_path.exists():
+        return []
+    files = [await f.url() async for f in await fs.listfiles(str(user_config_path), recursive=False)]
+    return [f for f in files if f.endswith('.ini')]
 
-    script_file = b.read_input(script_cloud_file)
-    config_file = b.read_input(user_config_cloud_file)
 
-    j.env('HAIL_QUERY_BACKEND', 'batch')
+async def get_volume_mount_files(fs: RouterAsyncFS, src: str) -> List[str]:
+    src = __real_absolute_local_path(src, strict=False)  # defer strictness checks and globbing
```

Comment:
There's another typing error here, given that `src` is a `str` rather than a path, either we need to convert to a string in `__real_absolute_local_path` or we need to do so here.